src/main.c

```diff
@@ -39,18 +96,37 @@ int main(int argc, char* argv[])
     ssize_t result;
     while (1)
     {
+        if (should_reload)
+        {
+            fprintf(stdout, "info: reloading\n");
+            release_input(eventPath);
+            readConfiguration();
+            if (bind_input(eventPath) != EXIT_SUCCESS)
+            {
+                fprintf(stderr, "error: could not capture the keyboard device\n");
+                return EXIT_FAILURE;
+            }
+            should_reload = 0;
+        }
+        if (should_exit)
+        {
+            fprintf(stdout, "info: exiting\n");
+            release_input(eventPath);
+            release_output();
+            return EXIT_SUCCESS;
+        }
         result = read(input, &inputEvent, sizeof(inputEvent));
-        if (result == (ssize_t) - 1)
+        if (result == (ssize_t) - 1 && errno == EINTR)
         {
-            if (errno == EINTR) continue;
+            continue;
         }
         if (result == (ssize_t)0)
         {
-            return ENOENT;
+            return EXIT_FAILURE;
         }
         if (result != sizeof(inputEvent))
         {
-            return EIO;
+            return EXIT_FAILURE;

```

Comment:
Should these exits try to release the devices?

Reply:
They should, and now do.